tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -298,3 +446,80 @@ def test_to_hierarchical_dataset_hierarchical_structure(
         expected_level_segments = hierarchical_structure_complex.get_level_segments(level_name=level_name)
         obtained_level_segments = hs.get_level_segments(level_name=level_name)
         assert sorted(obtained_level_segments) == sorted(expected_level_segments)
+
+
+@pytest.mark.parametrize(
```

Comment:
Also in the separate test use some fixtures for df and df_exog with regressors in the following cases:
1. df and df_exog on the same level and on different level after aggregation
2. df and df_exog on the different levels and on the same level after aggregation